modAionImpl/src/org/aion/zero/impl/sync/ResponseMgr.java

```diff
@@ -0,0 +1,98 @@
+package org.aion.zero.impl.sync;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class ResponseMgr {
+
+    private final Lock responseLock = new ReentrantLock();
+
+    /**
+     * Records time of requests/responses for active peer nodes
+     *
+     * @implNote Access to this resource is managed by the {@Link #responseLock}.
+     */
+    private final Map<String, LinkedList<Long>> requestTimeByPeers = new HashMap<>();
+
+    private final Map<String, Pair<Double, Integer>> responseStatsByPeers = new HashMap<>();
+
+    /**
+     * Log the time of request sent to an active peer node
+     *
+     * @param _nodeId peer node displya Id
+     * @param _requestTime time when the request was sent in nanoseconds
+     */
+    public void addPeerRequestTime(String _nodeId, long _requestTime) {
+        responseLock.lock();
+        try {
+            LinkedList<Long> requestStartTimes =
+                    requestTimeByPeers.containsKey(_nodeId)
+                            ? requestTimeByPeers.get(_nodeId)
+                            : new LinkedList<>();
+            requestStartTimes.add(_requestTime);
+            requestTimeByPeers.put(_nodeId, requestStartTimes);
```

Comment:
This operation is not necessary if the list already exists in the map. Try to refactor the code to avoid the `put` operation.